code/modules/cooking/machinery/cooking_machines/_appliance.dm

```diff
@@ -77,155 +80,108 @@
 		return
-
-	if (!Adjacent(usr) && !issilicon(usr))
-		to_chat(usr, "You can't adjust the [src] from this distance, get closer!")
-		return
-
-	if(output_options.len)
-		var/choice = input("What specific food do you wish to make with \the [src]?") as null|anything in output_options+"Default"
-		if(!choice)
-			return
-		if(choice == "Default")
-			selected_option = null
-			to_chat(usr, "<span class='notice'>You decide not to make anything specific with \the [src].</span>")
-		else
-			selected_option = choice
-			to_chat(usr, "<span class='notice'>You prepare \the [src] to make \a [selected_option] with the next thing you put in. Try putting several ingredients in a container!</span>")
+	selected_option = (choice == "Default") ? null : choice
+	to_chat(usr, SPAN_NOTICE("You decide to make [choice == "Default" ? "nothing specific" : choice] with [src]."))
 
 //Handles all validity checking and error messages for inserting things
 /obj/machinery/appliance/proc/can_insert(var/obj/item/I, var/mob/user)
-	if(!dropsafety(I))
-		return 0
+	if(issilicon(I.loc) || istype(I.loc, /obj/item/rig_module))
+		return FALSE
 
 	// We are trying to cook a grabbed mob.
 	var/obj/item/grab/G = I
 	if(istype(G))
 
 		if(!can_cook_mobs)
-			to_chat(user, "<span class='warning'>That's not going to fit.</span>")
-			return 0
+			to_chat(user, SPAN_WARNING("That's not going to fit."))
+			return FALSE
 
 		if(!isliving(G.affecting))
-			to_chat(user, "<span class='warning'>You can't cook that.</span>")
-			return 0
+			to_chat(user, SPAN_WARNING("You can't cook that."))
+			return FALSE
 
 		return 2
 
-
 	if (!has_space(I))
-		to_chat(user, "<span class='warning'>There's no room in [src] for that!</span>")
-		return 0
-
+		to_chat(user, SPAN_WARNING("There's no room in [src] for that!"))
+		return FALSE
 
-	if (container_type && istype(I, container_type))
-		return 1
+	if (istype(I, /obj/item/reagent_containers/cooking_container))
+		var/obj/item/reagent_containers/cooking_container/CC = I
+		if(CC.appliancetype & appliancetype)
+			return TRUE
 
 	// We're trying to cook something else. Check if it's valid.
 	var/obj/item/reagent_containers/food/snacks/check = I
 	if(istype(check) && islist(check.cooked) && (cook_type in check.cooked))
-		to_chat(user, "<span class='warning'>\The [check] has already been [cook_type].</span>")
-		return 0
+		to_chat(user, SPAN_WARNING("[check] has already been [cook_type]."))
+		return FALSE
 	else if(istype(check, /obj/item/reagent_containers/glass))
-		to_chat(user, "<span class='warning'>That would probably break [src].</span>")
-		return 0
-	else if(istype(check, /obj/item/disk/nuclear))
-		to_chat(user, "<span class='warning'>You can't cook that.</span>")
-		return 0
+		to_chat(user, SPAN_WARNING("That would probably break [src]."))
+		return FALSE
 	else if(I.iscrowbar() || I.isscrewdriver() || istype(I, /obj/item/storage/part_replacer))
-		return 0
+		return FALSE
 	else if(!istype(check) && !istype(check, /obj/item/holder))
```

Comment:
It totally does get reached. `!A && !B`= `!(A || B)` anyway.
`istype(check)` expands to `istype(check, /obj/item/reagent_containers/food/snacks)`.
`!istype(check, /obj/item/reagent_containers/food/snacks) && !istype(check, /obj/item/holder)` is completely valid.

Reply:
I see. I didn't check the original type, and assumed it was var/obj/item/check, because you then check if its type is a bunch of things that are *not* /obj/item/reagent_containers/food/snacks

check's definition should be as an /obj/item, then this line should be
```suggestion
	else if(!istype(check,/obj/item/reagent_containers/food/snacks) && !istype(check, /obj/item/holder))
```

Also please do not resolve things with an active conversation on them, I am having to open every single "resolved" conversation to check it, which is a waste of time over this many review comments. This is okay for trivial stuff if you *have* resolved it exactly as suggested, otherwise please leave them expanded for me.

Reply:
Still want this. My approval was premature... oops.